src/Package/Fetch.zig

```diff
@@ -59,6 +61,8 @@ actual_hash: Manifest.Digest,
 has_build_zig: bool,
 /// Indicates whether the task aborted due to an out-of-memory condition.
 oom_flag: bool,
+/// If `use_latest_commit` was true, this will be the commit that was used.
+latest_commit: ?git.Oid,
```

Comment:
This has an optional type, but is it ever `null`?
From a quick glance at the diff it seems to only be initialized to `undefined`, and only at one point assigned `want_oid`, which afaict is never be `null`. (It's possible I missed something though.)
If there's no logical state using the `null` value we could make the field non-optional.

Reply:
Nice catch!
I think (this was a month ago) it was made optional because not everything you can fetch is a git-repository, so it would not necessary get initialized. But you are right, it should _not_ be set to `undefined` everywhere, but `null`!